### DISCUSSION TITLE
websocket force closed on first incoming message

### DISCUSSION THREAD

**Discussion by buribalazs** [uNetworking/uWebSockets#935] [Q&A] [answered]
I tried to put together a simple test where a browser opens a page from localhost and sends a message to the server via ws. The handshake is fine and the connection opens. At the moment of the browser sending a message the connection abruptly closes.

This happens in Chrome, Firefox, IE Edge on Windows 10 and on Ubuntu Linux. Tried another WS library to check (https://www.npmjs.com/package/ws). That one didnt close the connection.

When the uWebSockets server sends a message, the connection stays alive and the message is delivered. It only doesnt work client -> server for me.

This is the whole "project":

server: nodejs v10.15.1
index.js (the server):
```
const fs = require('fs')
require("uWebSockets.js")
    .App()
    .ws("/*", {
        open: (ws, req) =>{
            console.log('open')

        },
        close: (ws,code,msg) => {
            console.log('close', code, msg)
        },
        message: (ws, message, isBinary) => {
            console.log(message)
        }
    })
    .get("/", (res, req) => {
        res.onAborted(a=>a)
        fs.readFile('src/app/index.html', (err,d)=>{
            res.writeHeader('Content-Type', 'text/html; charset=utf-8')
            .end(d)
        })
    })
    
    .listen(9001, listenSocket => {
        if (listenSocket) {
            console.log("Listening to port 9001")
        }
    })

```
index.html(the 'client'):
```
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>App</title>
</head>
<body>
    ok
    <script>
        let socket = new WebSocket('ws://localhost:9001');
        socket.onerror = e => console.log(e)
        socket.onmessage = e => console.log(e)
        socket.onopen = e => {
            console.log('open')
            socket.send('hello') // socket unexpectedly closed here with code 1006 this message is not delivered
        }
        socket.onclose = e => {
            console.log('closed', e)
        }
    </script>
</body>
</html>
```

**Comment by buribalazs** [comment, marked as answer]
@alexhultman after further investigation it looks like `maxPayloadLength` is mandatory, and if set, it works. I've set it to 16 and the server accepted the client's 'hello' message and the connection stayed open.

**Comment by ghost** [comment]
Ah, yes it defaults to 0 which is a pretty bad value.
